go/vt/schemamanager/tablet_executor.go

```diff
@@ -107,24 +107,22 @@ func (exec *TabletExecutor) Open(ctx context.Context, keyspace string) error {
 		return nil
 	}
 	exec.keyspace = keyspace
-	shardNames, err := exec.ts.GetShardNames(ctx, keyspace)
+	shards, err := exec.ts.FindAllShardsInKeyspace(ctx, keyspace, &topo.FindAllShardsInKeyspaceOptions{
+		Concurrency: topo.DefaultConcurrency, // Limit concurrency to avoid overwhelming the topo server.
+	})
```

Comment:
Nit picking here but we are defining a default `&topo.FindAllShardsInKeyspaceOptions` in three different places, perhaps we should create a global default variable and use that here and in the two other places.

Reply:
In this file, in `go/vt/topo/test/shard.go` and in `go/vt/vtctl/workflow/utils.go`

Reply:
https://github.com/vitessio/vitess/pull/15047/commits/8a0638e4b5130d8dd279791f7db504aac31843f0